src/transaction/transaction.rs

```diff
@@ -73,6 +82,7 @@ impl Transaction {
     /// # });
     /// ```
     pub async fn get(&self, key: impl Into<Key>) -> Result<Option<Value>> {
+        self.check_status()?;
```

Comment:
We could check this only on commit/rollback, i.e., the semantics is that we only allow one of those actions to happen once. Then we can combine the check and the set, and it makes things a bit less noisy. On the other hand we don't catch bugs where the user commits once but calls `get` after the commit, etc.

Reply:
Read after commit is valid but not expected, I think.
I feel like allowing operations after commit is a weird behavior, especially if users don't know how we implement it. They don't know whether the data is buffered locally or sent to the server. If their writes succeed but they cannot commit or rollback, they're going to be confused. 
Besides, checking before every operation may help users detect their errors early.

Reply:
Makes sense